From bd528253874b323a3fef5e2a5272190441fecbe4 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Thu, 27 Jun 2013 16:40:47 +0300
Subject: ARM: i.MX6: change the imx_add_ldb() function prototype

Change the function prototype in order to let the callers to pass a device ID as a parameter.
---
 arch/arm/mach-mx6/board-mx6q_arm2.c             |    2 +-
 arch/arm/mach-mx6/board-mx6q_sabreauto.c        |    2 +-
 arch/arm/mach-mx6/board-mx6q_sabrelite.c        |    2 +-
 arch/arm/mach-mx6/board-mx6q_sabresd.c          |    2 +-
 arch/arm/mach-mx6/devices-imx6q.h               |    4 ++--
 arch/arm/plat-mxc/devices/platform-imx_ldb.c    |    5 ++---
 arch/arm/plat-mxc/include/mach/devices-common.h |    2 +-
 7 files changed, 9 insertions(+), 10 deletions(-)

diff --git a/arch/arm/mach-mx6/board-mx6q_arm2.c b/arch/arm/mach-mx6/board-mx6q_arm2.c
index 97b601de..ef184d2 100644
--- a/arch/arm/mach-mx6/board-mx6q_arm2.c
+++ b/arch/arm/mach-mx6/board-mx6q_arm2.c
@@ -2151,7 +2151,7 @@ static void __init mx6_arm2_init(void)
 		imx6q_add_mipi_dsi(&mipi_dsi_pdata);
 	imx6q_add_vdoa();
 	imx6q_add_lcdif(&lcdif_data);
-	imx6q_add_ldb(&ldb_data);
+	imx6q_add_ldb(-1, &ldb_data);
 	imx6q_add_v4l2_output(0);
 	imx6q_add_v4l2_capture(0, &capture_data[0]);
 	imx6q_add_v4l2_capture(1, &capture_data[1]);
diff --git a/arch/arm/mach-mx6/board-mx6q_sabreauto.c b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
index fb590f5..d7e29e2 100644
--- a/arch/arm/mach-mx6/board-mx6q_sabreauto.c
+++ b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
@@ -1640,7 +1640,7 @@ static void __init mx6_board_init(void)
 	imx6q_add_vdoa();
 	imx6q_add_mipi_dsi(&mipi_dsi_pdata);
 	imx6q_add_lcdif(&lcdif_data);
-	imx6q_add_ldb(&ldb_data);
+	imx6q_add_ldb(-1, &ldb_data);
 	imx6q_add_v4l2_output(0);
 	imx6q_add_v4l2_capture(0, &capture_data[0]);
 	imx6q_add_v4l2_capture(1, &capture_data[1]);
diff --git a/arch/arm/mach-mx6/board-mx6q_sabrelite.c b/arch/arm/mach-mx6/board-mx6q_sabrelite.c
index e1075fe..1c8154d 100644
--- a/arch/arm/mach-mx6/board-mx6q_sabrelite.c
+++ b/arch/arm/mach-mx6/board-mx6q_sabrelite.c
@@ -1183,7 +1183,7 @@ static void __init mx6_sabrelite_board_init(void)
 
 	imx6q_add_vdoa();
 	imx6q_add_lcdif(&lcdif_data);
-	imx6q_add_ldb(&ldb_data);
+	imx6q_add_ldb(-1, &ldb_data);
 	imx6q_add_v4l2_output(0);
 	imx6q_add_v4l2_capture(0, &capture_data[0]);
 	imx6q_add_v4l2_capture(1, &capture_data[1]);
diff --git a/arch/arm/mach-mx6/board-mx6q_sabresd.c b/arch/arm/mach-mx6/board-mx6q_sabresd.c
index ac9b8f8..b858cb9 100644
--- a/arch/arm/mach-mx6/board-mx6q_sabresd.c
+++ b/arch/arm/mach-mx6/board-mx6q_sabresd.c
@@ -1746,7 +1746,7 @@ static void __init mx6_sabresd_board_init(void)
 	imx6q_add_vdoa();
 	imx6q_add_mipi_dsi(&mipi_dsi_pdata);
 	imx6q_add_lcdif(&lcdif_data);
-	imx6q_add_ldb(&ldb_data);
+	imx6q_add_ldb(-1, &ldb_data);
 	imx6q_add_v4l2_output(0);
 	imx6q_add_v4l2_capture(0, &capture_data[0]);
 	imx6q_add_v4l2_capture(1, &capture_data[1]);
diff --git a/arch/arm/mach-mx6/devices-imx6q.h b/arch/arm/mach-mx6/devices-imx6q.h
index f7ad317..6d901f8 100644
--- a/arch/arm/mach-mx6/devices-imx6q.h
+++ b/arch/arm/mach-mx6/devices-imx6q.h
@@ -120,8 +120,8 @@ extern const struct imx_ipuv3_data imx6q_ipuv3_data[] __initconst;
 			0, NULL, 0, pdata, sizeof(*pdata));
 
 extern const struct imx_ldb_data imx6q_ldb_data __initconst;
-#define imx6q_add_ldb(pdata) \
-	imx_add_ldb(&imx6q_ldb_data, pdata);
+#define imx6q_add_ldb(id, pdata) \
+	imx_add_ldb(&imx6q_ldb_data, pdata, id);
 
 #define imx6q_add_v4l2_output(id)	\
 	platform_device_register_resndata(NULL, "mxc_v4l2_output",\
diff --git a/arch/arm/plat-mxc/devices/platform-imx_ldb.c b/arch/arm/plat-mxc/devices/platform-imx_ldb.c
index a0919a7..dd5382e 100644
--- a/arch/arm/plat-mxc/devices/platform-imx_ldb.c
+++ b/arch/arm/plat-mxc/devices/platform-imx_ldb.c
@@ -40,7 +40,7 @@ const struct imx_ldb_data imx6q_ldb_data __initconst =
 
 struct platform_device *__init imx_add_ldb(
 		const struct imx_ldb_data *data,
-		struct fsl_mxc_ldb_platform_data *pdata)
+		struct fsl_mxc_ldb_platform_data *pdata, int id)
 {
 	struct resource res[] = {
 		{
@@ -50,7 +50,6 @@ struct platform_device *__init imx_add_ldb(
 		},
 	};
 
-	return imx_add_platform_device("mxc_ldb", -1,
+	return imx_add_platform_device("mxc_ldb", id,
 			res, ARRAY_SIZE(res), pdata, sizeof(*pdata));
 }
-
diff --git a/arch/arm/plat-mxc/include/mach/devices-common.h b/arch/arm/plat-mxc/include/mach/devices-common.h
index f6d616d..ba93d31 100755
--- a/arch/arm/plat-mxc/include/mach/devices-common.h
+++ b/arch/arm/plat-mxc/include/mach/devices-common.h
@@ -450,7 +450,7 @@ struct imx_ldb_data {
 
 struct platform_device *__init imx_add_ldb(
 	const struct imx_ldb_data *data,
-	struct fsl_mxc_ldb_platform_data *pdata);
+	struct fsl_mxc_ldb_platform_data *pdata, int id);
 
 struct imx_mxc_scc2_data {
 	resource_size_t iobase;
-- 
1.7.9.5

